### PR TITLE
Bug 1456878 - Support markdown comments

### PR DESCRIPTION
**Pull request by purelogiq**
### Each commit has a pretty-ok commit message to help understand what changed in that commit!

This PR adds support for markdown comments.

# Caveats! Please see!
- Midair collision page renders the inbound comment as plaintext.
- **FIXED** ~~User story field renders plaintext (can change if we want to, I just don't know how this field is used in practice).~~
- Markdown links don't have the 'nofollow' ref on them.
- **FIXED** ~~`> blockquotes` have the `>` even on a continued line (e.g. it'll look like `> line 1 > line 1 continued`). This is a bug caused by calling `html_quote` before `render_markdown` but changing that may mean changing the regex which I **really** don't want to do...thoughts?~~
- **FIXED** ~~At the moment it is only missing a little banner to inform people that all comments are now Markdown.~~

### CODE REVIEW

**Comment by purelogiq** [comment]
Video: https://drive.google.com/file/d/14ug9xXUGn8ks3eP42mlXhl9fwy8_Ok6r/view?usp=sharing

File: Bugzilla/Template.pm [hunk 1]
```diff
@@ -130,17 +130,19 @@ sub get_format {
...
 # module by Gareth Rees <garethr@cre.canon.co.uk>.  It has been heavily hacked,
 # all that is really recognizable from the original is bits of the regular
 # expressions.
 # This has been rewritten to be faster, mainly by substituting 'as we go'.
 # If you want to modify this routine, read the comments carefully
+# Renamed from 'quoteUrls' to 'renderComment' after markdown support was added.
 
-sub quoteUrls {
-    my ($text, $bug, $comment, $user, $bug_link_func) = @_;
+sub renderComment {
```
[the remark this answers is not in the document]
Comment (dylanwh) [reply Bugzilla/Template.pm]: Set $skip_markdown to 1 if !$comment->is_markdown, and simplify the other if statements.

**Comment by purelogiq** [comment]
# Pictures!

### New and improved blockquotes
double-block-quote


### Markdown is here notice
I went with a minimal approach. I think this is something we can just leave up. I've linked to the github docs since we use GFM.

On the show bug page
markdown-tip-comment

On the new bug page (same for new attachment)
markdown-tip-new-bug

File: template/en/default/email/bugmail.html.tmpl [hunk 2]
```diff
@@ -38,7 +38,7 @@
               on [% "$terms.bug $bug.id" FILTER bug_link(bug, { full_url => 1, user => to_user }) FILTER none %]
               from [% INCLUDE global/user.html.tmpl user = to_user, who = comment.author %]</b>
           [% END %]
-        <pre>[% comment.body_full({ wrap => 1 }) FILTER quoteUrls(bug, comment, to_user) %]</pre>
+        <pre>[% comment.body_full({ wrap => 1 }) FILTER renderComment(bug, comment) %]</pre>
```
Comment (dklawren) [inline template/en/default/email/bugmail.html.tmpl]: Don't we want markdown to render in HTML emails? The &lt;pre> tag would not allow this.

Comment (purelogiq) [reply template/en/default/email/bugmail.html.tmpl]: Thanks for catching this! I fixed it in the `extensions/BMO/template/en/default/email/bugmail.html.tmpl` but forgot to fix it here.

**Comment by purelogiq** [comment]
Looks better!

## Before (mono space, low line height)
before-font

## After (san serif, default line height)
even-after

**Comment by purelogiq** [comment]
even-even-after

**Comment by purelogiq** [comment]
@kyoshino True true, just wanted to get this out before the push to -dev :)

**Comment by dylanwh** [comment]
@kyoshino, @purelogiq I'm going to be doing some last-ditch testing but I think with three approvals this is good. Communication around this feature should be done so as to not surprise people too much.

**Pull request merged**